File: thirdSeminar/src/main/java/org/sopt/thirdSeminar/exception/handler/GlobalExceptionHandler.java
package org.sopt.thirdSeminar.exception.handler;

import org.sopt.thirdSeminar.common.dto.ErrorResponse;
import org.sopt.thirdSeminar.exception.ErrorMessage;
import org.sopt.thirdSeminar.exception.model.NotFoundException;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(IllegalArgumentException.class)
    public ErrorResponse handleIllegalArgumentException(final IllegalArgumentException e) {
        return ErrorResponse.error(ErrorMessage.BAD_REQUEST_EXCEPTION);
    }

    @ExceptionHandler(NotFoundException.class)
    public ErrorResponse handleNotFoundException(final NotFoundException e) {
        return ErrorResponse.error(e.getErrorMessage());
    }
}

File: sixthKotlin/src/main/kotlin/org/sopt/sixthKotlin/common/GlobalExceptionHandler.kt
package org.sopt.sixthKotlin.common

import org.sopt.sixthKotlin.common.dto.ErrorResponse
import org.sopt.sixthKotlin.exception.SoptException
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice


@RestControllerAdvice
class ControllerAdvice {
    @ExceptionHandler(SoptException::class)
    fun handleSoptException(e : SoptException): ErrorResponse {
        return ErrorResponse(e.error.status.value(), e.error.message);
    }
}